aten/src/ATen/native/quantized/fake_quant_per_channel_affine.cpp

```diff
@@ -35,6 +35,10 @@ Tensor fake_quantize_per_channel_affine(
     int64_t quant_min,
     int64_t quant_max) {
   TORCH_CHECK(self.scalar_type() == ScalarType::Float);
+  TORCH_CHECK(scale.scalar_type() == ScalarType::Float,
```

Comment:
double I think?

Reply:
I think it's float, as per the iterator expectation [here](https://github.com/pytorch/pytorch/blob/4cf81b32cb1aacb543980ae751b99aca22b3fc31/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp#L2037). Should I change them to double?

Reply:
Oh I thought this is quantize_per_tensor.

If this is float then probably zero_point should be int32_t, just to be consistent.

Reply:
The problem is that in the kernel implementation it is `float` for scale and `int64_t` for the zero_point. I will have to change all the kernels

Reply:
do you know why is that? it would be better to be consistent

Reply:
I am not sure why the inconsistency, but I see some kernels use 64 bits and some 32.